engine/runtime/src/main/java/org/enso/interpreter/runtime/scope/DebugLocalScope.java

```diff
@@ -188,15 +205,78 @@ boolean isMemberReadable(String memberName) {
   @ExportMessage
   @TruffleBoundary
   Object readMember(String member, @CachedLibrary("this") InteropLibrary interop)
-      throws UnknownIdentifierException {
+      throws UnknownIdentifierException, UnsupportedMessageException {
     if (!allBindings.containsKey(member)) {
       throw UnknownIdentifierException.create(member);
     }
     FramePointer framePtr = allBindings.get(member);
     var value = getValue(frame, framePtr);
+    if (value != null && isHostObject(value)) {
+      // Determine if the host object can be converted to corresponding Enso value.
+      var interopUncached = InteropLibrary.getUncached();
+      var conversionType = HostMethodCallNode.getPolyglotConversionType(value, interopUncached);
+      try {
+        switch (conversionType) {
+          case CONVERT_TO_DATE -> {
+            var localDate = interopUncached.asDate(value);
+            return new EnsoDate(localDate);
+          }
+          case CONVERT_TO_ARRAY -> {
+            return ArrayLikeHelpers.asVectorFromArray(value);
+          }
+          case CONVERT_TO_BIG_INT -> {
```

Comment:
I am still not sure why these conversion aren't in [EnsoLanguage.getLanguageView](https://github.com/enso-org/enso/blob/develop/engine/runtime/src/main/java/org/enso/interpreter/EnsoLanguage.java#L370)?

Because what is defined here is _Enso language view_ of these generic values. E.g. it should be in the `getLanguageView` method and this method can delegate to it, if necessary.

Reply:
[DebugValue](https://github.com/oracle/graal/blob/a4af3090d8788b4983c76b51b278bebbbeaa65c1/truffle/src/com.oracle.truffle.api.debug/src/com/oracle/truffle/api/debug/DebugValue.java#L1181) calls into the `getLanguageView` method - at least to handle some cases like _to string_.

Reply:
Polyglot/foreign object conversion functionality moved into `EnsoLanguage.getLanguageView` in 9134e528a7101c25606f9a57b8d4cff37c4bd558